### ISSUE TITLE
[Bug]: Warning: The selected contact is based on vcard version 2.1

### ISSUE THREAD

**Issue by troubas** [scheibler/khard#194]
Hi,

I'm always receiving this warning:

```
Warning:
The selected contact is based on vcard version 2.1 but khard only supports the creation and modification of vcards with version 3.0 and 4.0.
If you proceed, the contact will be converted to vcard version 3.0 but beware: This could corrupt the contact file or cause data loss.
```

However after confirming, validating the data and saving the vcard isn't updated to version 3.0 - I would like it to be updated. I'm not sure if this is an issue in khard or vobject?

**Comment by scheibler** [comment]
Khard only implements parts of the vcard standard for version 3.0 and 4.0. A full implementation is
much to costly at least for me. For example the "tel" (phone) and "adr" (post address) attributes
are supported however the "photo" attribute is not. Therefore a safe modification strategy is to
only add/modify/remove the known attributes and leave the others like "photo" be. So I do no harm to
things I don't understand. But that also means that I can't offer a full vcard converter.

The vcard version 2.1 is too old and not supported at all. If I encounter such a vcard file I
have three choices:

1. I try to read out what I can, present that in the modification editor and update all to let's say
vcard version 4.0. That might work for the supported attributes but what should I do with the
unsupported ones like "photo"? I can't simply delete them but copy them directly could be fatal as
well cause maybe the attribute structure changed massively from vcard version 2.1 to 4.0. In the end I could produce a vcard 4.0 file with 2.1 attributes.

2. I try to read out what I can and write the modifications back into the 2.1 file. Cause I only
support vcard version 3.0 and 4.0, I maybe end with a 2.1 file with 4.0 attributes. Not so much
better either. But at least I know the supported attributes and I've compared their 3.0 structure
manually to their 2.1 counterpart - in the most cases that should work.

3. No modification allowed at all. Safest way but hey ... no modification.

Therefore I decided to use method 2 in combination with a warning. But you are right the warning
text implies that khard will convert the 2.1 file into a 3.0 or 4.0 file. That's misleading and I
should update the warning text to make that clearer.
